docs/reference/modules/snapshots.asciidoc

```diff
@@ -329,6 +329,38 @@ POST /_snapshot/my_unverified_backup/_verify
 
 It returns a list of nodes where repository was successfully verified or an error message if verification process failed.
 
+[float]
+===== Repository Cleanup
+Repositories can over time accumulate data that is not referenced by any existing snapshot. This is a result of the data safety guarantees
+the snapshot functionality provides in failure scenarios during snapshot creation and the decentralized nature of the snapshot creation
+process. This unreferenced data does in no way negatively impact the performance or safety of a snapshot repository but leads to higher
+than necessary storage use. In order to clean up this unreferenced data, users can call the cleanup endpoint for a repository which will
+trigger a complete accounting of the repositories contents and subsequent deletion of all unreferenced data that was found.
+
+[source,js]
+-----------------------------------
+POST /_snapshot/my_repository/_cleanup
+-----------------------------------
+// CONSOLE
+// TEST[continued]
+
+The response to a cleanup request looks as follows:
+
+[source,js]
+--------------------------------------------------
+{
+  "results": {
```

Comment:
I think "results" is more natural language wise here in English (I guess because we're listing multiple non-binary values but I tbh. I don't have a good explanation as to why that is :D).

Reply:
It is the result of the clean up operation, so I think we could just drop the `results`/`result` field, maybe also the RepositoryCleanupResult object, and report 
```
{
    "deleted_bytes": 20,
    "deleted_blobs": 5
}
```

Reply:
I mainly did this to stay a little consistent in style with other snapshot APIs. We never really return a flat object, we always wrap under a descriptive key don't we? (e.g. getting a list of snapshots in a repo, creating a snapshot, ...)

Aesthetically I always kind of like an object like that but I don't have a good argument for it otherwise :) I suppose it makes it a little easier to extend if we ever want to in the future, but  that isn't so relevant for this endpoint I guess.